test/e2e/upg_e2e.go

```diff
@@ -604,6 +604,93 @@ var _ = ginkgo.Describe("CLI debug commands", func() {
 
 // TODO: validate both binapi and CLI against each other
 var _ = ginkgo.Describe("UPG Binary API", func() {
+	ginkgo.Context("for upf nat pool", func() {
+		f := framework.NewDefaultFramework(framework.UPGModeTDF, framework.UPGIPModeV4)
+
+		addPool := func(start string, end string, identity string, name string, isAdd bool) {
+			start_ip, _ := ip_types.ParseIP4Address(start)
+			end_ip, _ := ip_types.ParseIP4Address(end)
+
+			nwi := append(util.EncodeFQDN(identity), 0)
+			nameBytes := append([]byte(name), 0)
+
+			req := &upf.UpfNatPoolAdd{
+				IsAdd: isAdd,
+
+				MinPort:   2000,
+				MaxPort:   3000,
+				BlockSize: 1,
+				Start:     start_ip,
+				End:       end_ip,
+
+				Nwi:    nwi,
+				NwiLen: uint8(len(nwi)),
```

Comment:
NWI length should be calculated automatically by GoVPP, we don't need to specify it here